src/app/components/pages/guide-section-article-page/guide-section-article-page.component.ts

```diff
@@ -38,7 +38,7 @@ import {UserResolver} from '../../../resolvers/user/user.resolver';
         </container-card>
         <div>
           <guide-card
-            [fadedTitle]="(guideSectionArticle | async)?.translatedText?.title"
+            [fadedTitle]="(guideSectionArticle | async)?.section.translatedText?.title"
```

Comment:
Personally I don't really like having calls chained like this. 

article >> async >> section >> translated >> title (4-5 levels depending on how you count).

@bex1 is there a nice and Angular idiomatic way to extract this to its own function or property so we don't have to chain it like this?

Reply:
No which sucks.. | and the ?. operator is not available outside the template.
However you should change it to [fadedTitle]="(guideSectionArticle | async)?.section?.translatedText?.title since its safer to use ?. in the whole chain as a convention.

The only way to do it outside the template is to:
resolve the promise in loadData and save the result in another variable.

```
async loadData() {
  article = await guideSectionArticlePromise;
}
```

Then you would get article?.section?.translatedText?.title

If you use a function that would look like:

```
getArticleTitle() {
  return (article && article.section && article.section.translatedText && article.section.translatedText.title) || undefined;
}
```

Also In Angular, the dirty checking process has been optimized but functions called in component templates will still be called when any properties at the component level change, regardless of whether the function is dependent on those properties. This could lead to performance issues if used improperly.

Reply:
OK, thanks for a great explanation @bex1 👍 

Good catch 🎣 finding the missing `?.` :)

Reply:
I'll keep the current way of calling it for now.